### PR TITLE
Trigger states: restore from config file

### LINKED ISSUE #58: Trigger conditions for a newly started source are hard-coded, not restored from config
The loop in `data_source.go`, specifically in `func (ds *AnySource) PrepareRun()` sets various "sensible defaults" for the trigger conditions when a new source is started.

If the trigger settings have been saved into a viper configuration file, *these* ought to override any defaults. Also, the defaults ought to be all triggers off.

### PR DESCRIPTION
**Pull request by joefowler**
The trigger state was being saved to the config file but not restored from it. This is now done, substituting sensible defaults (all trigger types off) for channels not in the saved state.

Fixes #58

### CODE REVIEW

File: data_source.go [hunk 1]
```diff
@@ -337,6 +338,27 @@ func (ds *AnySource) PrepareRun() error {
...
+		for _, chnum := range ts.ChanNumbers {
+			if chnum < ds.nchan {
+				tsptrs[chnum] = &(fts[i].TriggerState)
+			}
+		}
+	}
+	// Use defaultTS for any channels not in the stored state.
+	// This will be needed any time you have more channels than in the
+	// last saved configuration.
+	defaultTS := TriggerState{AutoDelay: 250 * time.Millisecond,
```
Comment (ggggggggg) [inline data_source.go]: This should be all OFF! It's easy to turn them on, it's super frustrating to track down why a channel is triggering when I don't expect it to do so.

File: data_source.go [hunk 2]
```diff
@@ -345,19 +367,18 @@ func (ds *AnySource) PrepareRun() error {
...
+		dsp.DecimateLevel = 2
 		dsp.DecimateAvgMode = true
-		dsp.LevelTrigger = false
-		dsp.LevelLevel = 4000
-		dsp.EdgeTrigger = true
-		dsp.EdgeLevel = 100
-		dsp.EdgeRising = true
-		dsp.NPresamples = 200
+		dsp.NPresamples = 250
 		dsp.NSamples = 1000
```
Comment (ggggggggg) [inline data_source.go]: Can we save NSamples and NPresamples as well? Maybe in a different PR, but a keything to minimize steps for doing the same thing every time.

File: data_source.go [hunk 3]
```diff
@@ -345,19 +367,18 @@ func (ds *AnySource) PrepareRun() error {
...
+		if ts == nil {
+			ts = &defaultTS
+		}
+		dsp.TriggerState = *ts
+
+		// TODO: don't just set decimation and samples/presamples to arbitrary values
 		dsp.Decimate = false
-		dsp.DecimateLevel = 3
+		dsp.DecimateLevel = 2
 		dsp.DecimateAvgMode = true
```
Comment (ggggggggg) [inline data_source.go]: Why is Decimate on by default?

File: lancero/lancero.go [hunk 4]
```diff
@@ -193,6 +193,7 @@ func OdDashTX(b []byte, maxLines int) string {
 	outBuffer.WriteString("L = least significant byte, M = most significant byte\n")
 	outBuffer.WriteString("framebit in fbkL\n")
 	outBuffer.WriteString("errLerrM fbkLfdbM errLerrM fbkLfdbM errLerrM fbkLfdbM errLerrM fbkLfdbM \n")
```
Comment (ggggggggg) [inline lancero/lancero.go]: Are these right once the bits are flipped to make words read little endian?

File: data_source.go (hunk 1 above)
Comment (joefowler) [reply data_source.go]: This is OFF. But I can be more explicit, if you like.

File: data_source.go (hunk 3 above)
Comment (joefowler) [reply data_source.go]: It isn't:

`dsp.Decimate = false`

File: data_source.go (hunk 2 above)
Comment (joefowler) [reply data_source.go]: Actually, I think these ARE already saved. Let me figure this out...

File: data_source.go (hunk 3 above)
Comment (ggggggggg) [reply data_source.go]: But then why set DecimateLevel at all? Should it just be set to the zero value, and when someone turns it on they can set the values they want.

File: lancero/lancero.go (hunk 4 above)
Comment (joefowler) [reply lancero/lancero.go]: I'll check. I think they should read "fMfLeMeL"

I'll fix.

File: data_source.go [hunk 5]
```diff
@@ -355,9 +355,16 @@ func (ds *AnySource) PrepareRun() error {
...
+	// last saved configuration. All trigger types are disabled.
+	defaultTS := TriggerState{
+		AutoTrigger:  false,
+		AutoDelay:    250 * time.Millisecond,
+		EdgeTrigger:  false,
+		EdgeLevel:    100,
+		EdgeRising:   true,
+		LevelTrigger: false,
+		LevelLevel:   4000,
+	}
```
Comment (ggggggggg) [inline data_source.go]: I think this line should read `defaultTS := TriggetState{}`, I don't see why we should have any default nonzero values. What effect does this have? I guess it effectively sets the text in the dastard commander trigger ui?

It's good enough that I'll stop complaining about it and let you decide.

Comment (joefowler) [reply data_source.go]: Right: These other values set the text that you'll see in the GUI when you start to use it. They ensure that no nonsensical values appear there.

**Pull request merged**